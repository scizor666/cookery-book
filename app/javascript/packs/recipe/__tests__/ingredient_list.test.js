import React from 'react';
import IngredientList from '../ingredient_list';
import Ingredient from '../ingredient';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';

const ingredients = [
    {
        weight: 300,
        ingredientId: 1,
        name: 'chicken',
        productId: 2,
        caloricity: 107.7
    },
    {
        weight: 155,
        ingredientId: 2,
        name: 'carrot',
        productId: 1,
        caloricity: 44
    }];

test('ingredient list displayed', () => {
    const ingredientList = renderer.create(
        <IngredientList ingredients={ingredients}/>
    );
    let tree = ingredientList.toJSON();
    expect(tree).toMatchSnapshot();
});

test('new ingredient component added', () => {
    const ingredientList = shallow(
        <IngredientList ingredients={ingredients}/>
    );
    expect(ingredientList.find(Ingredient).length).toBe(2);

    ingredientList.find('#add_new_ingredient').simulate('click');
    expect(ingredientList.find(Ingredient).length).toBe(3);
});