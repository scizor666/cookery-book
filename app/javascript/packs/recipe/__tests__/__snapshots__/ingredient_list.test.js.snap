// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ingredient list displayed 1`] = `
<div>
  <div
    className="ingredient"
  >
    <label>
      <b>
        Ingredient
      </b>
    </label>
    <div
      className="row"
    >
      <div
        className="form-group col-md-4 col-sm-12"
      >
        <label
          htmlFor="recipe_product__name"
        >
          Name
        </label>
        <input
          autoComplete="off"
          className="form-control product-name"
          data-name="name"
          id="recipe_product__name"
          name="recipe[product][][name]"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="name"
          required="required"
          type="text"
          value="chicken"
        />
      </div>
      <div
        className="form-group col-md-4 col-sm-12"
      >
        <label
          htmlFor="recipe_product__caloricity"
        >
          Caloricity, kcal/100g
        </label>
        <input
          autoComplete="off"
          className="form-control product-caloricity"
          data-name="caloricity"
          id="recipe_product__caloricity"
          min="0"
          name="recipe[product][][caloricity]"
          onChange={[Function]}
          placeholder="caloricity"
          required="required"
          step="any"
          type="number"
          value={107.7}
        />
        <input
          className="product-id"
          defaultValue={2}
          id="recipe_product__id"
          name="recipe[product][][id]"
          type="hidden"
        />
      </div>
      <div
        className="form-group col-md-4 col-sm-12"
      >
        <label
          htmlFor="recipe_ingredient__weight"
        >
          Weight, g
        </label>
        <input
          autoComplete="off"
          className="form-control ingredient-weight"
          data-name="weight"
          id="recipe_ingredient__weight"
          min="0"
          name="recipe[ingredient][][weight]"
          onChange={[Function]}
          placeholder="weight"
          required="required"
          step="any"
          type="number"
          value={300}
        />
        <input
          defaultValue={1}
          id="recipe_ingredient__id"
          name="recipe[ingredient][][id]"
          type="hidden"
        />
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col"
      >
        <a
          className="remove-ingredient"
          href="javascript:void(0);"
          onClick={[Function]}
          style={
            Object {
              "display": "block",
            }
          }
        >
          <i
            className="fa fa-minus fa-2x pull-right"
            style={
              Object {
                "color": "red",
              }
            }
          />
        </a>
      </div>
    </div>
  </div>
  <div
    className="ingredient"
  >
    <label>
      <b>
        Ingredient
      </b>
    </label>
    <div
      className="row"
    >
      <div
        className="form-group col-md-4 col-sm-12"
      >
        <label
          htmlFor="recipe_product__name"
        >
          Name
        </label>
        <input
          autoComplete="off"
          className="form-control product-name"
          data-name="name"
          id="recipe_product__name"
          name="recipe[product][][name]"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="name"
          required="required"
          type="text"
          value="carrot"
        />
      </div>
      <div
        className="form-group col-md-4 col-sm-12"
      >
        <label
          htmlFor="recipe_product__caloricity"
        >
          Caloricity, kcal/100g
        </label>
        <input
          autoComplete="off"
          className="form-control product-caloricity"
          data-name="caloricity"
          id="recipe_product__caloricity"
          min="0"
          name="recipe[product][][caloricity]"
          onChange={[Function]}
          placeholder="caloricity"
          required="required"
          step="any"
          type="number"
          value={44}
        />
        <input
          className="product-id"
          defaultValue={1}
          id="recipe_product__id"
          name="recipe[product][][id]"
          type="hidden"
        />
      </div>
      <div
        className="form-group col-md-4 col-sm-12"
      >
        <label
          htmlFor="recipe_ingredient__weight"
        >
          Weight, g
        </label>
        <input
          autoComplete="off"
          className="form-control ingredient-weight"
          data-name="weight"
          id="recipe_ingredient__weight"
          min="0"
          name="recipe[ingredient][][weight]"
          onChange={[Function]}
          placeholder="weight"
          required="required"
          step="any"
          type="number"
          value={155}
        />
        <input
          defaultValue={2}
          id="recipe_ingredient__id"
          name="recipe[ingredient][][id]"
          type="hidden"
        />
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col"
      >
        <a
          className="remove-ingredient"
          href="javascript:void(0);"
          onClick={[Function]}
          style={
            Object {
              "display": "block",
            }
          }
        >
          <i
            className="fa fa-minus fa-2x pull-right"
            style={
              Object {
                "color": "red",
              }
            }
          />
        </a>
      </div>
    </div>
  </div>
  <div
    className="row mt-4"
  >
    <div
      className="col"
    >
      <a
        href="javascript:void(0)"
        id="add_new_ingredient"
        onClick={[Function]}
      >
        <i
          className="fa fa-plus fa-2x text-success pull-right top-aligned-row"
        />
      </a>
    </div>
  </div>
</div>
`;
